platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -92,8 +93,16 @@ export default class HangingProtocolService extends PubSubService {
     },
     isReconstructable: {
       name: 'Checks if the display set is reconstructable',
-      // we can add more advanced checking here
-      callback: displaySet => displaySet.isReconstructable ?? false,
+      // We need to check again if display set is reconstructable in case we've
+      // got some new metadata in the instances since the last time we've check
+      callback: displaySet => {
+        const {
+          value: isReconstructable,
+          missingFrames: missingFrames,
+        } = isDisplaySetReconstructable(displaySet.instances);
+
+        return isReconstructable && missingFrames == 0
```

Comment:
I think an idea would be to fire and event in the dicomMetadataStore when an instance is updated and maybe recalculate the displaySet.isReconstructable in an event handler? 

This will most likely break the dicomweb or if not break would slow it down